DotNetHowToEncryptionUsingCMK/DotNetHowToEncryptionUsingCMK/Program.cs

```diff
@@ -0,0 +1,329 @@
+            return indexClient;
+        }
+
+        private static void CleanupResources(SearchServiceClient serviceClient)
+        {
+            if (serviceClient.Indexes.Exists("hotels"))
+            {
+                serviceClient.Indexes.Delete("hotels");
+            }
+
+            if (serviceClient.SynonymMaps.Exists("desc-synonymmap"))
+            {
+                serviceClient.SynonymMaps.Delete("desc-synonymmap");
+            }
+        }
+
+        private static void CreateSynonymsEncryptedUsingCustomerManagedKey(SearchServiceClient serviceClient, IConfigurationRoot configuration)
+        {
+            var synonymMap = new SynonymMap()
+            {
+                Name = "desc-synonymmap",
+                Synonyms = "hotel, motel\ninternet,wifi\nfive star=>luxury\neconomy,inexpensive=>budget",
+                EncryptionKey = GetEncryptionKeyFromConfiguration(configuration)
+            };
+
+            serviceClient.SynonymMaps.CreateOrUpdate(synonymMap);
+        }
+
+        private static void CreateHotelsIndexEncryptedUsingCustomerManagedKey(SearchServiceClient serviceClient, IConfigurationRoot configuration)
+        {
+            var definition = new Index()
+            {
+                Name = "hotels",
+                Fields = FieldBuilder.BuildForType<Hotel>(),
+                EncryptionKey = GetEncryptionKeyFromConfiguration(configuration)
+            };
+
+            serviceClient.Indexes.Create(definition);
+        }
+
+        private static EncryptionKey GetEncryptionKeyFromConfiguration(IConfigurationRoot configuration)
+        {
+            Uri keyVaultKeyUri = new Uri(configuration["AzureKeyVaultKeyUrl"]);
+            if (!keyVaultKeyUri.Host.Contains("vault.azure.net") || keyVaultKeyUri.Segments.Length != 4 || keyVaultKeyUri.Segments[1] != "keys/")
+            {
+                throw new ArgumentException("Invalid 'AzureKeyVaultKeyUrl' - Expected format: 'https://<key-vault-name>.vault.azure.net/keys/<key-name>/<key-version>'", "AzureKeyVaultKeyUrl");
+            }
+
+            var encryptionKey = new EncryptionKey
+            {
+                KeyVaultUri = $"{keyVaultKeyUri.Scheme}://{keyVaultKeyUri.Host}",
+                KeyVaultKeyName = keyVaultKeyUri.Segments[2].Trim('/'),
+                KeyVaultKeyVersion = keyVaultKeyUri.Segments[3].Trim('/')
+            };
+
+            string applicationId = configuration["AzureActiveDirectoryApplicationId"];
+            if (!string.IsNullOrWhiteSpace(applicationId))
+            {
+                encryptionKey.AccessCredentials = new AzureActiveDirectoryApplicationCredentials
+                {
+                    ApplicationId = applicationId,
+                    ApplicationSecret = configuration["AzureActiveDirectoryApplicationSecret"]
+                };
+            }
+
+            encryptionKey.Validate();
+            return encryptionKey;
+        }
+
+#if HowToExample
```

Comment:
The conditional compilation in the original HowTo project is only there because that part of the code does double-duty as an example for two different articles. You can just pick one approach and delete the `#if`.